server/src/main/java/io/mantisrx/server/master/domain/DataFormatAdapter.java

```diff
@@ -352,14 +356,21 @@ public static JobWorker convertMantisWorkerMetadataWriteableToMantisWorkerMetada
                 .withSlaveID(writeable.getSlaveID())
                 .withStageNum(writeable.getStageNum())
 
-                .withState(convertMantisJobStateToWorkerState(writeable.getState()))
                 .withWorkerIndex(writeable.getWorkerIndex())
                 .withWorkerNumber(writeable.getWorkerNumber())
 
                 .withJobCompletedReason(writeable.getReason())
                 .withPreferredCluster(writeable.getCluster())
-                .withLifecycleEventsPublisher(eventPublisher)
-                .build();
+                .withLifecycleEventsPublisher(eventPublisher);
+
+        if (workerPorts == null && !MantisJobState.isTerminalState(writeable.getState())) {
+            logger.info("marking worker {} for Job ID {} as {}", writeable.getWorkerId(), jobId, WorkerState.Failed);
```

Comment:
can also add a JobcompletedReason when marking it FAILED.. 
Maybe, even add a metric that we could alert on it if this happens.

Reply:
I thought about that and setting `completedAt` as well. But is the job really completed? There could be other workers which could be fine.

Reply:
The enum is confusing due to legacy reasons, it is used for both worker and job IIRC. In this case, it only applies to the worker and not the job.. If it is making this PR complex, feel free to leave it out and address later

Reply:
Oh interesting. No problem, adding completed reason (`Killed`) is a simple change. I'll do that.